### PR TITLE
define fill! methods for datalayouts

### LINKED ISSUE #861: BoundsError in `implicit_tendency!` function
## Describe the bug
During the investigation of the usage of 3rd-order upwinding with small aspect-ratio elements, I discovered a bug in the `implicit_tendnecy!` defined in `examples/hybrid/`. Here is the full stack trace, if helpful. 

[main_bounds_error_stack_trace.log](https://github.com/CliMA/ClimaCore.jl/files/9254172/main_bounds_error_stack_trace.log)

## To Reproduce
In the `main` branch, from ClimaCore.jl top-level, use the following command-line
```julia
julia --project=examples --check-bounds=yes -e 'ENV["FLOAT_TYPE"]="Float64";  ENV["TEST_NAME"]="plane/inertial_gravity_wave"; include("examples/hybrid/driver.jl")'
```

<details><summary>Project</summary>
<p>
Using the examples Project.toml from `main`.

</p>
</details>

## System details

Any relevant system information:
- Julia version: 1.7.2
- operating system: Ubuntu 20.04

### PR DESCRIPTION
**Pull request by simonbyrne**
Should hopefully fix #861

- [x] Code follows the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) OR N/A.
- [x] Unit tests are included OR N/A.
- [x] Code is exercised in an integration test OR N/A.
- [x] Documentation has been added/updated OR N/A.

### CODE REVIEW

File: src/DataLayouts/DataLayouts.jl [hunk 1]
```diff
@@ -574,6 +591,12 @@ end
 function Base.size(data::IJF{S, Nij}) where {S, Nij}
     return (Nij, Nij, 1, 1, 1)
 end
+function Base.fill!(data::IJF{S, Nij}, val) where {S, Nij}
+    for j in 1:Nij, i in 1:Nij
```
Comment (charleskawczynski) [inline src/DataLayouts/DataLayouts.jl:595]:
```suggestion
    @inbounds for j in 1:Nij, i in 1:Nij
```
We should still add `@inbounds` here, right?

**Comment by valeriabarra** [comment]
bors try

**Review by charleskawczynski — APPROVED** [review]

**Pull request merged**